### PR TITLE
fix(nixos/sddm): use the kde sddm package

### PR DESCRIPTION
**Pull request by isabelroses**
This resolves the error where the theme doesn't work if plasma6 isn't enabled. by default the packge is qt5 sddm, but catppuccin/sddm uses qt6.

### CODE REVIEW

**Comment by isabelroses** [comment]
closes https://github.com/catppuccin/nix/issues/229

**Review by getchoo — CHANGES_REQUESTED** [review]
i think this would be better as an assertion...but i'm not really sure how to reliably do that as sddm doesn't directly report its qt's major version

my first thoughts are asserting that pkgs.kdePackages.qtbase is in the buildInputs of config.services.displayManager.sddm.package or making a conditionally applied assertion for services.displayManager.sddm.package via a new option

**Comment by isabelroses** [comment]
I think the easiest thing would be to check what package is being used since it's either the qt5 one or the kde package.

**Comment by isabelroses** [comment]
or perhaps `lib.mkDefault` would be better so we don't get clashes with users that apply the plasma 6 desktop.

**Comment by getchoo** [comment]
> I think the easiest thing would be to check what package is being used since it's either the qt5 one or the kde package.

that's what i meant with a conditional assertion. it wouldn't account for overriden packages ofc, but if someone really knows what they're doing they can disable it with an option

> or perhaps lib.mkDefault would be better

i don't think it would as users could easily set the qt5 package as normal with no warning. assertions on the other hand provide a built in explanation of why we need to use the qt6 version when this happens

> so we don't get clashes with users that apply the plasma 6 desktop

why would this happen?

**Comment by isabelroses** [comment]
> why would this happen?

The duplicate package deceleration I mean.

File: modules/nixos/sddm.nix [hunk 1]
```diff
@@ -55,5 +56,12 @@ in
           ;
       })
     ];
+
+    assertions = [
+      {
+        assertion = config.services.displayManager.sddm.package != pkgs.kdePackages.sddm;
+        message = "Not using the kdePackages.sddm package, will cause the theme to fail being applied";
+      }
+    ];
```
Comment (getchoo) [inline modules/nixos/sddm.nix:65]:
```suggestion
    assertions = lib.optional cfg.assertQt6Sddm {
      assertion = config.services.displayManager.sddm.package != pkgs.kdePackages.sddm;
      message = "Not using the kdePackages.sddm package, will cause the theme to fail being applied";
    };
```

`assertQt6Sddm` should probably be
```nix
{
  assertQt6Sddm = lib.mkEnableOption "checking `services.displayManager.sddm.package` is the Qt 6 version. This is to ensure the theme is applied properly, but may have false positives in the case of overridden packages for example" // { default = true; };
}
```
(hopefully with better formatting :trollface:)

**Review by getchoo — CHANGES_REQUESTED** [review]

File: modules/nixos/sddm.nix (hunk 1 above)
Comment (isabelroses) [reply modules/nixos/sddm.nix:65]: wow that's cursed but sure.

**Review by getchoo — APPROVED** [review]
thanks!

**Pull request merged**